parcels/particleset/particlesetsoa.py

```diff
@@ -427,6 +478,38 @@ def to_dict(self, pfile, time, deleted_only=False):
         return self._collection.toDictionary(pfile=pfile, time=time,
                                              deleted_only=deleted_only)
 
+    def compute_neighbor_tree(self, time, dt):
+        active_mask = self.active_particles_mask(time, dt)
+
+        self._values = np.vstack((
+            self._collection.data['lat'],
+            self._collection.data['lon'],
+            self._collection.data['depth'],
+        ))
+        if self._dirty_neighbor:
+            self._neighbor_tree.rebuild(self._values, active_mask=active_mask)
+            self._dirty_neighbor = False
+            self._neighbor_time = time
+        else:
+            self._neighbor_tree.update_values(self._values, new_active_mask=active_mask)
+
+    def neighbors_by_index(self, particle_idx):
+        neighbor_idx, distances = self._neighbor_tree.find_neighbors_by_idx(
+            particle_idx)
+        neighbor_idx = self._active_particle_idx[neighbor_idx]
+        mask = (neighbor_idx != particle_idx)
+        neighbor_idx = neighbor_idx[mask]
+        if 'surf_dist' in self._collection.data:
```

Comment:
that is a rather unusual check - could please rather use the `ptype` to check if an attribute is part of the ParticleType description ? It is what we used until now in all the rest of the code.

Reply:
Can you check if I have done it right now?